### PR TITLE
Scrub PSP docs for 1.25

### PR DESCRIPTION
**Pull request by liggitt**
PodSecurityPolicy admission is dropped in 1.25 in https://github.com/kubernetes/kubernetes/pull/109798

This updates docs for 1.25 to indicate removal and truncates the PSP docs that aren't oriented at migration.

/sig auth
/cc @deads2k @tallclair

### CODE REVIEW

File: content/en/docs/concepts/security/pod-security-policy.md
Comment (tallclair) [inline content/en/docs/concepts/security/pod-security-policy.md:469]: Do you think the policy reference could be useful to people looking to recreate their PSPs elsewhere?

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (tallclair) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:702]:
```suggestion
This is the replacement for the removed `PodSecurityPolicy` admission controller.
```
or "former"?

File: content/en/docs/concepts/security/pod-security-standards.md
Comment (tallclair) [inline content/en/docs/concepts/security/pod-security-standards.md:459]: I think you can delete these example files. https://github.com/kubernetes/website/tree/main/content/en/examples/policy

File: content/en/docs/concepts/security/pod-security-policy.md
Comment (liggitt) [reply content/en/docs/concepts/security/pod-security-policy.md:469]: historical content will be available at https://v1-24.docs.kubernetes.io/ … not sure we should keep info from non-current versions at HEAD

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:702]: I think I prefer “former”. Or:
```suggestion
Older versions of Kubernetes included an admission controller for
the beta PodSecurityPolicy API; the Pod Security admission controller
provides similar, but not identical, security enforcement.
```

Comment (sftim) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:702]: I think I prefer “former”. Or:
```suggestion
Older versions of Kubernetes included an admission controller for
the beta PodSecurityPolicy API; the Pod Security admission controller
provides similar, but not identical, security enforcement.
```

File: content/en/docs/reference/glossary/pod-security-policy.md
Comment (liggitt) [reply content/en/docs/reference/glossary/pod-security-policy.md:20]: reworded

File: content/en/docs/concepts/security/pod-security-standards.md
Comment (liggitt) [reply content/en/docs/concepts/security/pod-security-standards.md:459]: I'll leave this alone for now based on @sftim's comment

**Comment by liggitt** [comment]
addressed comments in a separate commit

**Comment by sftim** [comment]
I suggest waiting for  #34675 to merge **and** to appear on the _dev-1.25_ branch before we approve these changes - that way, there's less risk of missing a detail.

**Comment by sftim** [comment]
https://github.com/kubernetes/website/pull/34675 has merged, so the next resync to _dev-1.25_ should include that change.

File: content/en/docs/concepts/security/pod-security-admission.md
Comment (shannonxtreme) [inline content/en/docs/concepts/security/pod-security-admission.md:24]:
```suggestion
to the `PodSecurityPolicy` API, which was removed in Kubernetes v1.25. Pod security restrictions
```

File: content/en/docs/concepts/security/pod-security-admission.md
Comment (shannonxtreme) [inline content/en/docs/concepts/security/pod-security-admission.md:30]: If you add the previous suggestion, you can remove this note

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (shannonxtreme) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:710]:
```suggestion
the beta `PodSecurityPolicy` API; the `PodSecurity` admission controller
```

File: content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md
Comment (shannonxtreme) [inline content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md:12]:
```suggestion
`PodSecurityPolicy` objects, whether the field mutates
```

**Review by shannonxtreme — APPROVED** [review]
Little nits

**Comment by cathchu** [comment]
Hey @liggitt Docs Shadow here 👋 ! There seems to be some conflicts in this branch that need to be resolved before this PR can be merged, as well as some comments that have not been resolved. Please take a look when you can so we can merge!

**Comment by liggitt** [comment]
looks like https://github.com/kubernetes/website/pull/35618 merged in the meantime... not sure how much of this PR still applies

**Comment by liggitt** [comment]
rebased and applied formatting suggestions

**Pull request merged**
